satpy/readers/seviri_l1b_native.py

```diff
@@ -193,10 +193,27 @@ def __init__(self, filename, filename_info, filetype_info,
         # Available channels are known only after the header has been read
         self.header_type = get_native_header(has_archive_header(self.filename))
         self._read_header()
-        self.dask_array = da.from_array(self._get_memmap(), chunks=(CHUNK_SIZE,))
+        self._make_dask_array_with_map_blocks()
         self._read_trailer()
         self.image_boundaries = ImageBoundaries(self.header, self.trailer, self.mda)
 
+    def _make_dask_array_with_map_blocks(self):
+        """Makes the dask array using the ``da.map_blocks()`` functionality."""
+        dtype = self._get_data_dtype()
+        chunks = da.core.normalize_chunks(
+            "auto",
+            shape=(self.mda["number_of_lines"],),
+            dtype=dtype)
+        self.dask_array = da.map_blocks(
+            _get_array,
+            dtype=dtype,
+            chunks=chunks,
+            meta=np.zeros(1, dtype=dtype),
```

Comment:
I think the dask docs usually suggest `meta=np.array((), dtype=dtype)` instead. I believe you using `zeros` makes numpy do more work than this suggestion.

Reply:
I just replaced `np.zeros` with `meta=np.array([], dtype=dtype)`. One needs to use `[]`  instead of `()` as the first argument of `np.array`. Otherwise, we run into issues with broadcasting the shapes given the record nature of the `dtype`.